### PR TITLE
feat: add support for api token authentication without cloudId

### PR DESCRIPTION
**Pull request by lucianaparaschivei**
We can use an api token authentication without the cloud connection pool.

#### Checklist
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org)
- [ ] documentation is updated

### CODE REVIEW

File: src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs [hunk 1]
```diff
@@ -275,7 +276,20 @@ protected override void InitializeTarget()
             {
                 var username = _username?.Render(eventInfo) ?? string.Empty;
                 var password = _password?.Render(eventInfo) ?? string.Empty;
-                config = config.BasicAuthentication(username, password);
+
+                if (!string.IsNullOrEmpty(username) && !string.IsNullOrEmpty(password))
```
Comment (snakefoot) [inline src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs:280]: Think you should mark `RequireAuth` as obsolete and no longer depend on it. Instead just check username like this:

`if (!IsNullOrWhiteSpace(username))`

Comment (lucianaparaschivei) [reply src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs]: done

File: src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs [hunk 2]
```diff
@@ -245,38 +246,45 @@ protected override void InitializeTarget()
...
-                connectionPool = new StaticConnectionPool(nodes);  
+                connectionPool = new StaticConnectionPool(nodes);
             }
 
             var config = ElasticsearchSerializer == null
                 ? new ConnectionConfiguration(connectionPool)
                 : new ConnectionConfiguration(connectionPool, ElasticsearchSerializer);
 
-            if (RequireAuth)
+            if (!string.IsNullOrWhiteSpace(username))
```
Comment (snakefoot) [inline src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs:277]: Guess one need to replace `if (RequireAuth)` with `if (string.IsNullOrWhiteSpace(cloudId))` so it becomes:

```c#
if (string.IsNullOrWhiteSpace(cloudId))
{
            if (!string.IsNullOrWhiteSpace(username))
            {
                config = config.BasicAuthentication(username, password);
            }
            else
            {
                if (!string.IsNullOrWhiteSpace(apiKey) && !string.IsNullOrWhiteSpace(apiKeyId))
                {
                    config = config.ApiKeyAuthentication(apiKeyId, apiKey);
                }
            }
}
```

Comment (lucianaparaschivei) [reply src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs]: makes sense :)

**Review by snakefoot — APPROVED** [review]
Looks good to me

**Pull request merged**